### DISCUSSION TITLE
Error while processing response coming from the CNC

### DISCUSSION THREAD

**Discussion by cebess** [winder/Universal-G-Code-Sender#1722] [Help] [answered]
Recently I smoked the motherboard of my 3018 CNC. I purchased a replacement. I performed about 5-6 cuts just fine.

Now every once in a while, my run has errors. Usually the run stops at this point, but sometimes it will continue. Some examples from the log are:
example 1: The CNC is cutting a contour with a ramp and it receives the move command and generates this error but continues on:
**>>> X-14.866Y9.613Z-7.329
ok
[Error] Error while processing response <<Run|MPos:-16.349,10.000,-7.309|FS:100,1000>C ok>: For input string: "1000>C o"**

Example 2: This is another error on the same run that stopped the tun:
**>>> X-2.652Y3.199Z-7.534
<Run|MPos:-6.998,5.210,-     M 100)    j
ok

[Error] Error while processing response <<Run|MPos:-3.773,15. Sbj r    M 100,1000>>**

Anyone have an idea what is going on? I have resorted to rebooting both the raspberry pi and the CNC between runs to see if it makes a difference, but have not seen any issue. I could have a bad motherboard that has some kind of thermal issue, a communications issue or some settings not correct.

The firmware settings on the CNC are:
**>>> $$
$0 = 10    (Step pulse time, microseconds)
$1 = 25    (Step idle delay, milliseconds)
$2 = 0    (Step pulse invert, mask)
$3 = 2    (Step direction invert, mask)
$4 = 0    (Invert step enable pin, boolean)
$5 = 0    (Invert limit pins, boolean)
$6 = 0    (Invert probe pin, boolean)
$10 = 1    (Status report options, mask)
$11 = 0.010    (Junction deviation, millimeters)
$12 = 0.002    (Arc tolerance, millimeters)
$13 = 0    (Report in inches, boolean)
$20 = 0    (Soft limits enable, boolean)
$21 = 0    (Hard limits enable, boolean)
$22 = 0    (Homing cycle enable, boolean)
$23 = 0    (Homing direction invert, mask)
$24 = 25.000    (Homing locate feed rate, mm/min)
$25 = 500.000    (Homing search seek rate, mm/min)
$26 = 250    (Homing switch debounce delay, milliseconds)
$27 = 1.000    (Homing switch pull-off distance, millimeters)
$30 = 1000    (Maximum spindle speed, RPM)
$31 = 0    (Minimum spindle speed, RPM)
$32 = 0    (Laser-mode enable, boolean)
$100 = 800.000    (X-axis travel resolution, step/mm)
$101 = 800.000    (Y-axis travel resolution, step/mm)
$102 = 800.000    (Z-axis travel resolution, step/mm)
$110 = 1000.000    (X-axis maximum rate, mm/min)
$111 = 1000.000    (Y-axis maximum rate, mm/min)
$112 = 600.000    (Z-axis maximum rate, mm/min)
$120 = 30.000    (X-axis acceleration, mm/sec^2)
$121 = 30.000    (Y-axis acceleration, mm/sec^2)
$122 = 30.000    (Z-axis acceleration, mm/sec^2)
$130 = 200.000    (X-axis maximum travel, millimeters)
$131 = 200.000    (Y-axis maximum travel, millimeters)
$132 = 200.000    (Z-axis maximum travel, millimeters)**

Looking at the error in Java UGS I see:
**java.lang.NullPointerException
	at com.willwinder.ugs.nbm.visualizer.shared.GcodeRenderer.setMachineCoordinate(GcodeRenderer.java:185)
	at com.willwinder.ugs.nbm.visualizer.RendererInputHandler.statusStringListener(RendererInputHandler.java:340)
	at com.willwinder.universalgcodesender.AbstractController.dispatchStatusString(AbstractController.java:863)
	at com.willwinder.universalgcodesender.GrblController.handleStatusString(GrblController.java:647)
[catch] at com.willwinder.universalgcodesender.GrblController.rawResponseHandler(GrblController.java:239)
	at com.willwinder.universalgcodesender.AbstractController.rawResponseListener(AbstractController.java:834)
	at com.willwinder.universalgcodesender.AbstractCommunicator.sendEventToListeners(AbstractCommunicator.java:172)
	at com.willwinder.universalgcodesender.AbstractCommunicator.lambda$new$0(AbstractCommunicator.java:194)
	at java.base/java.lang.Thread.run(Thread.java:830)**

I am running UGS 2.0.7 from Dec 2, 2020 on a Pi 4 and have not had issues with it before now.

**Comment by breiler** [comment]
It looks like you are getting interference when communicating with the controller board. Make sure your USB-cable is far away from any high power cables and that it is a properly shielded one, preferably one with those ferrite clumps on it.

**Reply by cebess** [reply]
It is shielded with chokes. One thing I am going to try is turning off arcs on the g code build to see if simpler g code will make a difference.

**Reply by cebess** [reply]
I replaced my cable with one that has chokes on both ends and I still get the errors. I am beginning to think that it really does have something to do with arcs, since I ran some jobs with no arcs and they were fine. Could there be a buffer overflow error while the controller is executing the arc?

**Comment by khturnings** [comment]
If you have replaced the motherboard with an original is have you checked if it is using the latest grbl firmware?

What version are you running?

**Comment by cebess** [comment]
It says it is running GRBL 1.1f

**Reply by khturnings** [reply]
My guess is that is ok - I am new to this too so will let the experts make more suggestions. Sorry I can't help, maybe post an example of the gcode if you think the arc's are causing the issue then someone can test that arc's are does not an issue.

**Reply by spip01** [reply]
I'm getting the same error. tried all the v2 releases. tried 3 cables. included the error message and the detailed message below. It's a new aurdino nano running grbl 1.1h.  i'm stuck with a project to deliver. 

this is while it's sitting here idle. It's a steady stream of these errors,

<Idle|MPos:0.000,0.000,20.320|Bf:15,128|FS:0,0|Pn:XYZ
[Error] Error while processing response <<Idl|MPos:0.000,0.000,20.320|Bf:15,128|FS:0,0Pn:XYZ|WCO:0.000,0.000,.000>>: For input string: "0Pn:XYZ"
[Error] Error while processing response <<Idle|MPos:0.0000.000,20.320|Bf:15,128|FS:0,0|Pn:XYZ>>
[Error] Error while processing response <<Idle|MPos:0.000,0.000,20.320|Bf:15,128|FS:00|Pn:XYZ|Ov:100,100,100>>: 1
[Error] Error while processing response <<Idle|MPos:0.000,0.000,20.320|Bf:15,128|FS:,0|Pn:XYZ>>: empty String
[Error] Error while processing response <<Idle|MPos:0.0000.000,20.320|Bf:15,128|FS:0,0|Pn:XYZ>>
[Error] Error while processing response <<Idle|MPos:0.000,0.00020.320|Bf:15,128|FS:0,0|Pn:XYZ|WCO:0.000,0.000,0.000>>
--------

this in the detail window

java.lang.NullPointerException
	at com.willwinder.ugs.nbm.visualizer.shared.GcodeRenderer.setMachineCoordinate(GcodeRenderer.java:183)
	at com.willwinder.ugs.nbm.visualizer.RendererInputHandler.statusStringListener(RendererInputHandler.java:338)
	at com.willwinder.universalgcodesender.AbstractController.dispatchStatusString(AbstractController.java:866)
	at com.willwinder.universalgcodesender.GrblController.handleStatusString(GrblController.java:656)
[catch] at com.willwinder.universalgcodesender.GrblController.rawResponseHandler(GrblController.java:240)
	at com.willwinder.universalgcodesender.AbstractController.rawResponseListener(AbstractController.java:837)
	at com.willwinder.universalgcodesender.AbstractCommunicator.sendEventToListeners(AbstractCommunicator.java:172)
	at com.willwinder.universalgcodesender.AbstractCommunicator.lambda$new$0(AbstractCommunicator.java:194)
	at java.lang.Thread.run(Unknown Source)

**Reply by breiler** [reply]
@spip01 there is not much we can do in UGS if you have a bad connection to the hardware.

Just as an example:
```<Idl|MPos:0.000,0.000,20.320|Bf:15,128|FS:0,0Pn:XYZ|WCO:0.000,0.000,.000>``` is missing an "e" in Idle and a "|" between "FS:0,0Pn:XYZ".

These types of errors can not be fixed in software. So you need to recheck your wiring and try to find the root cause for the bad connection. I had similar problems caused by EMI from my spindle, I got a properly shielded power cable for the spindle which resolved most of my problems.

**Reply by cebess** [reply]
It turns out the new motherboard was bad. I told the vendor. They sent a replacement. It has been working flawlessly ever since.

**Reply by breiler** [reply]
That was unfortunate, but I'm happy that you found the reason and the fix for it!

**Comment by cebess** [comment, marked as answer]
Mother board was bad.

**Comment by ghalandar66** [comment]
I also have this problem
But I replaced the chip and it gives an error again

Please send me the link if I need to install a certain version of Java and ugs

**Reply by breiler** [reply]
Assuming that it is the same error as above - Java and UGS version will not matter. It is a hardware error.

If you want more help than just my assumption, start a new discussion with more details than just "I also have this problem".

**Reply by ghalandar66** [reply]
ok

I tried the same hardware on my friend's computer, it worked without any problem
I even did it with another chip and it failed again

**Reply by ghalandar66** [reply]
Another problem is that the z-axis microswitch does not work And the movement of this axis is the opposite
I used arduino board and cnc shield and tb6600 driver for my station

And this is the software settings

$0 = 10    (step pulse, usec)
$1 = 25    (step idle delay, msec)
$2 = 0    (step port invert mask:00000000)
$3 = 0    (dir port invert mask:00000000)
$4 = 0    (step enable invert, bool)
$5 = 0    (limit pins invert, bool)
$6 = 0    (probe pin invert, bool)
$10 = 3    (status report mask:00000011)
$11 = 1.000    (junction deviation, mm)
$12 = 1.000    (arc tolerance, mm)
$13 = 0    (report inches, bool)
$20 = 0    (soft limits, bool)
$21 = 1    (hard limits, bool)
$22 = 1    (homing cycle, bool)
$23 = 232    (homing dir invert mask:11101000)
$24 = 1500.000    (homing feed, mm/min)
$25 = 500.000    (homing seek, mm/min)
$26 = 250    (homing debounce, msec)
$27 = 1.000    (homing pull-off, mm)
$100 = 320.000    (x, step/mm)
$101 = 320.000    (y, step/mm)
$102 = 320.000    (z, step/mm)
$110 = 800.000    (x max rate, mm/min)
$111 = 800.000    (y max rate, mm/min)
$112 = 800.000    (z max rate, mm/min)
$120 = 50.000    (x accel, mm/sec^2)
$121 = 50.000    (y accel, mm/sec^2)
$122 = 25.000    (z accel, mm/sec^2)
$130 = 200.000    (x max travel, mm)
$131 = 200.000    (y max travel, mm)
$132 = 200.000    (z max travel, mm)

**Reply by breiler** [reply]
Start a new discussion thread and provide details. 

Problem descriptions such as `z-axis microswitch does not work` contains too little information for us to work with. Describe what you tried, your expected result and the actual outcome.

**Reply by ghalandar66** [reply]
I need a complete tutorial for ugs software settings
So that I can describe my problem in more detail
Because I have no expertise in programming
I am a simple carpenter who decided to make my own cnc machine
But now I found many problems

**Reply by ghalandar66** [reply]
I think I need to install a special version of ugs and java for my computer

Which is the only version you suggest?

**Reply by breiler** [reply]
Please start a new discussion thread...

There are tons of guides and tutorials, but I can't point you to right direction without knowing the basics. What hardware do you have (names, links, photos)? What firmware are you using (GRBL 0.9, GRBL 1.1, GrblMega, GrblMega5x, other)?

Right now you have told us that you are running an Arduino and a CNC shield. Some shields requires GRBL 0.9 instead of the latest GRBL 1.1. Your GRBL settings is missing the `$32` which tells me you might be running GRBL 0.9. 🤷🏻‍♂️ 

You told us that you have problems with `z-axis microswitch does not work`. But not what the actual problem is. Is it triggering in the UI by directly after you've connected? Are you trying to home and it doesn't trigger? How is it connected? Is it `Normally open` or `Normally closed`? Photos or videos are extremely helpful. 

You are asking for advice on version for Java and UGS. The UGS Platform 2.0.20 is the latest and I think that you should run that: https://github.com/winder/Universal-G-Code-Sender#downloads

**Reply by ghalandar66** [reply]
ok

https://github.com/winder/Universal-G-Code-Sender/discussions/2312
